internal/auth/oidc/client_secret.go

```diff
@@ -0,0 +1,22 @@
+package oidc
+
+import (
+	"encoding/json"
+)
+
+// ClientSecret equals an AuthMethod's client secret.  This type provides a
+// wrapper so the secret isn't inadvertently leaked into a log or error.
+type ClientSecret string
+
+// RedactedClientSecret is the redacted string or json for an oidc id_token.
+const redactedClientSecret = "[REDACTED: OIDC client_secret]"
+
+// String will redact the client_secret.
+func (s ClientSecret) String() string {
+	return redactedClientSecret
+}
+
+// MarshalJSON will redact the client_secret.
+func (s ClientSecret) MarshalJSON() ([]byte, error) {
```

Comment:
For my own education, when does our code marshal these values?

Reply:
We currently don't but if someone adds a future log msg that does, this will redact the secret appropriately.